heron/ckptmgr/src/java/com/twitter/heron/ckptmgr/CheckpointManager.java

```diff
@@ -0,0 +1,241 @@
+    IStatefulStorage checkpointsBackend;
+    String classname = (String) checkpointManagerConfig.getCkptMgrConfig(
+                                             CheckpointManagerConfig.STORAGE_CLASSNAME);
+    try {
+      checkpointsBackend = (IStatefulStorage) Class.forName(classname).newInstance();
+    } catch (InstantiationException e) {
+      throw new CheckpointManagerException(classname + " class must have a no-arg constructor.", e);
+    } catch (IllegalAccessException e) {
+      throw new CheckpointManagerException(classname + " class must be concrete.", e);
+    } catch (ClassNotFoundException e) {
+      throw new CheckpointManagerException(classname + " class must be a class path.", e);
+    }
+    checkpointsBackend.init(
+        Collections.unmodifiableMap(checkpointManagerConfig.getBackendConfig()));
+
+    // Start the server
+    this.checkpointManagerServer = new CheckpointManagerServer(
+        topologyName, topologyId, checkpointMgrId, checkpointsBackend,
+        checkpointManagerServerLoop, serverHost, serverPort, serverSocketOptions);
+  }
+
+  public void start() {
+    // The CheckpointManagerServer would run in the main thread
+    // We do it in the final step since it would await the main thread
+    LOG.info("Starting CheckpointManager Server");
+    checkpointManagerServer.start();
+    checkpointManagerServerLoop.loop();
+  }
+
+  public static void main(String[] args) throws IOException,
+                                       ParseException, CheckpointManagerException {
+    Options options = constructOptions();
+    Options helpOptions = constructHelpOptions();
+    CommandLineParser parser = new DefaultParser();
+    // parse the help options first.
+    CommandLine cmd = parser.parse(helpOptions, args, true);
+
+    if (cmd.hasOption("h")) {
+      usage(options);
+      return;
+    }
+
+    try {
+      // Now parse the required options
+      cmd = parser.parse(options, args);
+    } catch (ParseException e) {
+      usage(options);
+      throw new RuntimeException("Error parsing command line options: ", e);
+    }
+
+    String topologyName = cmd.getOptionValue("topologyname");
+    String topologyId = cmd.getOptionValue("topologyid");
+    String ckptmgrId = cmd.getOptionValue("ckptmgrid");
+    int port = Integer.parseInt(cmd.getOptionValue("ckptmgrport"));
+    String stateConfigFilename = cmd.getOptionValue("ckptmgrconfig");
+    String heronInternalConfig = cmd.getOptionValue("heroninternalconfig");
+    SystemConfig systemConfig = SystemConfig.newBuilder(true).putAll(heronInternalConfig,
+                                                                     true).build();
+    CheckpointManagerConfig ckptmgrConfig = new CheckpointManagerConfig(stateConfigFilename);
+
+    // Add the SystemConfig into SingletonRegistry
+    SingletonRegistry.INSTANCE.registerSingleton(SystemConfig.HERON_SYSTEM_CONFIG, systemConfig);
```

Comment:
Again this is consistent with how Instance/MetricsManager/StMgr handle config. I would suggest to keep this consistent across different services and change all services in one go if we do this change.